block/pruning.go

```diff
@@ -5,12 +5,6 @@ import (
 )
 
 func (m *Manager) pruneBlocks(retainHeight uint64) (uint64, error) {
-	syncTarget := m.SyncTarget.Load()
-
-	if retainHeight > syncTarget {
```

Comment:
@danwt why u removed this check?

Reply:
It's not needed as far as I can tell, it's just an extra branch for no reason
I added a test to confirm

Reply:
needed for sequencer mode